core-java-modules/core-java-concurrency-basic-2/src/main/java/com/baeldung/concurrent/parking/ParkingThreadDemo.java

```diff
@@ -0,0 +1,47 @@
+package com.baeldung.concurrent.parking;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.*;
+
+public class ParkingThreadDemo {
+    private static final Logger LOG = LoggerFactory.getLogger(ParkingThreadDemo.class);
+
+    public void parkingThread() throws InterruptedException {
+        final ExecutorService executorPool = Executors.newFixedThreadPool(5);
+        final LinkedBlockingQueue<Long> queue = new LinkedBlockingQueue<Long>();
+        Collection<Future<Long>> collection = new ArrayList<Future<Long>>();
+
+        // consumer:
+        for (int i = 0; i < 20; i++) {
+            collection.add(executorPool.submit(new Callable<Long>() {
+                @Override
+                public Long call() throws Exception {
+                    while (true) {
+                        Long item = queue.take();
+                        if (item.intValue() == 0) {
+                            break;
```

Comment:
In this code, each of the 20 consumers will stop on hitting a zero. This is not incredibly easy to follow. This code is also quite monolithic and deeply nested. It is very abstract and seems to serve no real purpose.

Can we make a simpler more practical example?